### DISCUSSION TITLE
Prisma Migrate: Execute TS alongside SQL files?

### DISCUSSION THREAD

**Discussion by marek-hanzal** [prisma/prisma#13703] [Q&A] [answered]
Hello,

I love to use Prisma, but one thing I'm heavily missing is an ability to run a migration file generated by Prisma (so SQLs) and custom TS script instead of SQLs:
- SQL do database migration magick, for example create new columns with NULLable values
- TS script computes those values (I don't want to rewrite some pieces of code to SQL's updates - one source of truth in my app code)
- Another SQL which will mark the original NULLable columns as not NULLable; now I can to, because I've precomputed those required values

It's possible with Prisma to execute custom TS (or okay, JS) scripts as part of migration? Now I'm thinking about switching from Prisma Migrate to another tool which has this ability, but does not have my trust; also the workflow would be crazy as I have to use Prisma for SQL magick and the other tool for running those SQL with TS files and managing migrations with it. Crazy. I want Prisma :).

Thanks a lot for an answer!

P.S. This is a general question, but I'm creating a new project using latest Prisma.

**Comment by austincrim** [comment, marked as answer]
Hey @marek-hanzal 👋,

This isn't possible in Prisma today. The idea has come up in the past ([existing issue](https://github.com/prisma/prisma/issues/4688)) and it is something we are still considering.

We would love to get your unique perspective on that existing issue (and feel free to add a +1 to help us gauge interest).

**Reply by marek-hanzal** [reply]
Hello! Thank you for an answer, I've posted some ideas there, so we'll see. But it's quite old issue, I'm not sure if even anybody from Prisma cares about that issue...

**Reply by austincrim** [reply]
We definitely do care! We have a lot we want to accomplish in the future of the ORM and it's impossible to have the capacity for it all, but the feature requests with more activity are more likely to get prioritized sooner. We really value your input 🙏.

**Reply by marek-hanzal** [reply]
Ok, I didn't want to sound rude - sometimes it's unfortunatelly like that - you have an idea and it goes in the air.

So, I've found some way, how to workaround Prisma for now in a "compatible" way which means the solution *could* work later on if Prisma adds support for executing TS/JS stuff; I've wrote it down to that issue, so if anybody will be curious...

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
